lib/Controller/DocumentController.php

```diff
@@ -233,25 +236,32 @@ public function index($fileId, $dir) {
 	 * @PublicPage
 	 */
 	public function public($shareToken, $fileId) {
-		// If type of fileId is a string, then it
-		// doesn't work for shared documents, lets cast to int everytime
-		$fileId = \is_numeric($fileId) ? (int)$fileId : null;
+		if (\is_string($shareToken) && \strlen($shareToken) > 0 && \is_numeric($fileId)) {
```

Comment:
checks should match with the ones in the "index" method. Not sure if these are the good ones or not.

Reply:
@jvillafanez why so? these methods are different
```
public function public($shareToken, $fileId)
public function index($fileId, $dir)
```

Reply:
my bad. I didn't read the parameters correctly.